frappe/model/db_query.py

```diff
@@ -550,6 +556,75 @@ def build_filter_conditions(self, filters, conditions, ignore_permissions=None):
 			else:
 				conditions.append(self.prepare_filter_condition(f))
 
+	def apply_fieldlevel_read_permissions(self):
+		"""Apply fieldlevel read permissions to the query"""
+		if self.flags.ignore_permissions:
+			return
+
+		permitted_fields = get_permitted_fields(doctype=self.doctype)
+
+		for i, field in enumerate(self.fields):
+			if "distinct" in field.lower():
+				# field: 'count(distinct `tabPhoto`.name) as total_count'
+				# column: 'tabPhoto.name'
+				self.distinct = True
+				column = field.split(" ", 2)[1].replace("`", "").replace(")", "")
+			else:
+				# field: 'count(`tabPhoto`.name) as total_count'
+				# column: 'tabPhoto.name'
+				column = field.split("(")[-1].split(")", 1)[0]
+				column = strip_alias(column).replace("`", "")
+
+			if column == "*" and not in_function("*", field):
+				self.fields[i : i + 1] = permitted_fields
+				continue
+
+			# handle pseudo columns
+			elif not column or column.isnumeric():
+				continue
+
+			# labels / pseudo columns or frappe internals
+			elif column[0] in {"'", '"', "_"}:
+				continue
+
+			# handle child / joined table fields
+			elif "." in field:
+				table, column = column.split(".", 1)
+				ch_doctype = table.replace("`", "").replace("tab", "", 1)
+
+				if wrap_grave_quotes(table) in self.tables:
+					permitted_child_table_fields = get_permitted_fields(
+						doctype=ch_doctype, parenttype=self.doctype
+					)
+					if column in permitted_child_table_fields:
+						continue
+					else:
+						self.fields.remove(field)
+				else:
+					raise frappe.PermissionError(ch_doctype)
+
+			elif column in permitted_fields:
+				continue
+
+			# field inside function calls / * handles things like count(*)
+			elif "(" in field:
+				if "*" in field:
+					continue
+				elif any(x for x in permitted_fields if x in field):
```

Comment:
seems problematic. what happens if I have a field called "apple" which is permitted, but "apple_pie" which isn't?

Reply:
Removed this clause via https://github.com/frappe/frappe/pull/19533/commits/e41f005daad1da250524f31e03a808bbf5ac485e